### ISSUE TITLE
[Bug] download issue

### ISSUE THREAD

**Issue by shadzpw** [microsoft/playwright#17335]
**Context:**
- Playwright Version: [1.25.1]
- Operating System: [Windows]
- Node.js version: [14.17..6]
- Browser: [ Chromium]

**Code Snippet**
const [ download ] = await Promise.all([
  // Start waiting for the download
  page.waitForEvent('download'),
  // Perform the action that initiates download
  page.locator('.exportButton').click(),
]);
// Wait for the download process to complete
console.log(await download.path());

I am facing issue while downloading an excel (.xlsx format), exporting excel from the ag-grid.  Getting the below error:- page.waitForEvent: Timeout 30000ms exceeded while waiting for event "download"  I followed the same steps as described in https://playwright.dev/docs/downloads.  Note: I can see the "download" event being triggered by running the same test in Cypress Test Runner console and Test seems to be running fine in Cypress.
**Describe the bug**

Add any other details about the problem here.

**Comment by mxschmitt** [comment]
Can you provide us a repro which we can run locally? Otherwise we unfortunately can't act on it.

**Comment by shadzpw** [comment]
Owing to the privacy of the project, I won't be able to share with you all the details. Can you please tell me what do you need without comprising the privacy of the project?

**Comment by mxschmitt** [comment]
A script which we can run locally to reproduce your described issue.

**Comment by shadzpw** [comment]
const [ download ] = await Promise.all([
page.waitForEvent('download'),
page.locator('.exportButton').click(),
]);
console.log(await download.path());

Here '.exportButton' is the css locator of the Export button which exports excel from the ag grid.

**Comment by mxschmitt** [comment]
Maybe the wrong button? Maybe no download happening when clicking on that button?

**Comment by shadzpw** [comment]
![image](https://user-images.githubusercontent.com/113522161/190198695-984870e2-fed3-4ed0-bd54-39e45f051444.png)

**Comment by shadzpw** [comment]
I can see download happening when clicking on that button, so can't be a wrong button

**Comment by mxschmitt** [comment]
This is good, then there should be definitely be a download event dispatched! Maybe already earlier? Sorry without knowing the site its just guessing!

**Comment by shadzpw** [comment]
Could you/your team please guide with the next steps

**Comment by shadzpw** [comment]
Cypress successful run logs for the same Test:
![SAVE_20220914_214833](https://user-images.githubusercontent.com/113522161/190208359-0817a219-7a1e-4855-931e-53c4883bfa78.jpg)

**Comment by mxschmitt** [comment]
We can't provide next steps without knowing how your site behaves. As I said probably your download does emit already before clicking on the button. The code here https://github.com/microsoft/playwright/issues/17335#issuecomment-1246911051 looks good.

**Comment by shadzpw** [comment]
I tried a similar excel download scenario on an external website - **https://exceljet.net/excel-pivot-tables** and I got the same error:
page.waitForEvent: Timeout 30000ms exceeded while waiting for event "download"

Is the download event getting blocked on my network?
Any pointers which can help?

**Comment by mxschmitt** [comment]
The following does work for me, how did you try it?

```ts
  await page.goto('https://exceljet.net/excel-pivot-tables');
  const [download] = await Promise.all([
    page.waitForEvent('download'),
    page.click('text=sample data for pivot table.xlsx'),
  ]);
  console.log('Filename: ' + download.suggestedFilename());
  console.log('Path: ' + await download.path());
```

**Comment by shadzpw** [comment]
Yes running behind an enterprise corporate environment, using Windows 10 Enterprise, Version: 20H2

**Comment by mxschmitt** [comment]
Good find! Could you go to `chrome://policy` with your Chrome and see if you have entries there? One of them might be causing it. Just to confirm, you are using Chromium, not Chrome right?

A more technical debugging would be to go to PowerShell, set the `DEBUG=pw:protocol,pw:browser` env var, to see if we find the reason why its blocked there. You can do it via:

```
$Env:DEBUG="pw:protocol,pw:browser"
npx playwright test debug.spec.ts
```

and create the file debug.spec.ts

```ts
import { test, expect } from '@playwright/test';

test('test', async ({ page }) => {
  await page.goto('https://exceljet.net/excel-pivot-tables');
  const [download] = await Promise.all([
    page.waitForEvent('download'),
    page.click('text=sample data for pivot table.xlsx'),
  ]);
  console.log('Filename: ' + download.suggestedFilename());
  console.log('Path: ' + await download.path());
});
```

**Comment by shadzpw** [comment]
Yes, using Chromium
![IMG_20220916_230520](https://user-images.githubusercontent.com/113522161/190699457-3cc9af17-ca58-49d1-80c9-3e4a928f1b5e.jpg)

**Comment by shadzpw** [comment]
chrome://policy
![IMG_20220916_231221](https://user-images.githubusercontent.com/113522161/190700486-e7758cb1-d914-4369-8f34-f4be47600caf.jpg)

**Comment by mxschmitt** [comment]
Could you provide us your playwright config?

**Comment by shadzpw** [comment]
![Uploading IMG_20220917_100945.jpg…]()

**Comment by shadzpw** [comment]
![IMG_20220917_100945](https://user-images.githubusercontent.com/113522161/190840769-35cb7b7e-8844-4be8-aa29-f6b21ae56812.jpg)

**Comment by mxschmitt** [comment]
Could you do that what I've written here (https://github.com/microsoft/playwright/issues/17335#issuecomment-1249247239) about setting the environment variables? This should give us more hints whats going wrong. For that open a PowerShell Terminal or do it inside Visual Studio Code and execute the lines above.

**Comment by mxschmitt** [comment]
Closing for now as part of the triage process, if you still encounter issues, please follow https://github.com/microsoft/playwright/issues/17335#issuecomment-1249247239 and create a new issue.
